crowbar_framework/app/helpers/installer/upgrades_helper.rb

```diff
@@ -0,0 +1,67 @@
+
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Installer
+  module UpgradesHelper
+    def continue_button
+      button_tag icon_tag(:chevron_right, t(".continue")), class: "btn btn-primary"
+    end
+
+    def check_repos_button
+      if check_repos
+        continue_button
+      else
+        button_tag icon_tag(:refresh, t(".recheck")), class: "btn btn-primary"
+      end
+    end
+
+    def check_ha_repo
+      return nil unless Proposal.where(barclamp: "pacemaker")
+      return false unless Crowbar::Repository.provided?("ha")
+
+      unless Crowbar::Repository.provided_and_enabled?("ha")
+        Openstack::Upgrade.enable_repos_for_feature("ha", logger)
+      end
+
+      true
+    end
+
+    def check_ceph_repo
+      return nil unless Proposal.where(barclamp: "ceph")
+      return false unless Crowbar::Repository.provided?("ceph")
+
+      unless Crowbar::Repository.provided_and_enabled?("ceph")
+        Openstack::Upgrade.enable_repos_for_feature("ceph", logger)
+      end
+
+      true
+    end
+
+    def check_repos
+      check_ha_repo != false && check_ceph_repo != false
+    end
+
+    def database_node
+      @node ||= NodeObject.find("state:crowbar_upgrade AND run_list_map:database-server").first
```

Comment:
OK, so with https://github.com/crowbar/crowbar-core/pull/232 , this could be used to find the real node. I'm not sure if it is correct (had no chance to test it) but it might be
``@node = NodeObject.find("crowbar_upgrade_db_dumped_here:true")`